# Sample Jupyter Notebook

GitHub renders `.ipynb` files with **full output** — including plots, tables, and rich output.

This makes notebooks ideal for:
- Data analysis reports
- Tutorials with runnable code
- Scientific papers with embedded results

In [1]:
# Standard library imports
import math
import statistics
from collections import Counter

print('Libraries loaded.')

Libraries loaded.


## Git Commit Analysis

Let's analyze a sample dataset of commit frequencies.

In [2]:
# Sample commit data (commits per day over 4 weeks)
commits_per_day = [
    3, 7, 2, 5, 8, 1, 0,   # Week 1
    4, 6, 3, 9, 5, 2, 1,   # Week 2
    8, 4, 6, 7, 3, 0, 2,   # Week 3
    5, 9, 4, 6, 8, 3, 1,   # Week 4
]

print(f'Total commits    : {sum(commits_per_day)}')
print(f'Average per day  : {statistics.mean(commits_per_day):.1f}')
print(f'Median per day   : {statistics.median(commits_per_day)}')
print(f'Busiest day      : {max(commits_per_day)} commits')
print(f'Quietest day     : {min(commits_per_day)} commits')
print(f'Std deviation    : {statistics.stdev(commits_per_day):.2f}')

Total commits    : 132
Average per day  : 4.7
Median per day   : 4.5
Busiest day      : 9 commits
Quietest day     : 0 commits
Std deviation    : 2.65


## Conclusion

This notebook demonstrates how GitHub renders Jupyter notebooks.

| Stat | Value |
|------|-------|
| Total commits | 132 |
| Daily average | 4.7 |
| Peak day | 9 commits |

> GitHub renders notebooks **read-only**. To run cells, open in Colab or Jupyter.